test/plugins/test-trace-hapi.js

```diff
@@ -308,6 +308,41 @@ describe('hapi', function() {
           });
         });
       });
+
+      it('should end spans when client aborts request', function(done) {
```

Comment:
So I've found that what makes hapi and koa different from express + others is when `res.end` is called. From what I've seen digging into each of these codebases, hapi and koa call `res.end` after all middleware is done, but the function from which this is called short-circuits if the socket is already closed. On the other hand, express calls `res.end` in a call to `res.send`, a method that applications are meant to call within their middleware. This call to `res.end`, as far as I can tell, is guaranteed.